File: Folicon_Native/Model/ProIcon.vb
Imports System.Drawing
Imports System.IO
Imports FoliconNative.Modules

Public Class ProIcon
    Dim _filePath As string

    Public Sub New(filePath As string)
        _filePath = filePath
    End Sub

    Public Function RenderToBitmap() As Bitmap
        Return RenderTargetBitmapTo32bppArgb(AsRenderTargetBitmap())
    End Function

    Private Function AsRenderTargetBitmap() As BitmapSource
        using img As New Bitmap(_filePath)
            using icon = New Bitmap(img, 256, 256)
                Return LoadBitmap(icon)
            End Using
        End Using
    End Function

    Private Function RenderTargetBitmapTo32bppArgb(rtb As BitmapSource) As Bitmap
        Dim stream As New MemoryStream()
        Dim encoder As BitmapEncoder = New PngBitmapEncoder()
        encoder.Frames.Add(BitmapFrame.Create(rtb))
        encoder.Save(stream)
        Return New Bitmap(stream) 'png; 
    End Function
End Class

File: FoliCon/Models/ProIcon.cs
using System.Drawing;
using System.IO;
using System.Windows.Media.Imaging;

namespace FoliCon.Modules
{
    public class ProIcon
    {
        private string _filePath;

        public ProIcon(string filePath)
        {
            _filePath = filePath;
        }

        public Bitmap RenderToBitmap()
        {
            return RenderTargetBitmapTo32bppArgb(AsRenderTargetBitmap());
        }

        private BitmapSource AsRenderTargetBitmap()
        {
            using (Bitmap img = new Bitmap(_filePath))
            {
                using (var icon = new Bitmap(img, 256, 256))
                {
                    return Util.LoadBitmap(icon);
                }
            }
        }

        private Bitmap RenderTargetBitmapTo32bppArgb(BitmapSource rtb)
        {
            MemoryStream stream = new MemoryStream();
            BitmapEncoder encoder = new PngBitmapEncoder();
            encoder.Frames.Add(BitmapFrame.Create(rtb));
            encoder.Save(stream);
            return new Bitmap(stream); //png;
        }
    }

}
